src/Symfony/Component/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * A factory to instantiate the best possible HTTP client for the current runtime.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @experimental in 4.3
+ */
+final class HttpClient
+{
+    /**
+     * @param array $defaultOptions     Default requests' options
+     * @param int   $maxHostConnections The maximum number of connections to a single host
+     */
+    public static function create(array $defaultOptions = [], int $maxHostConnections = 6): HttpClientInterface
+    {
+        if (\extension_loaded('curl')) {
+            return new CurlHttpClient($defaultOptions, $maxHostConnections);
+        }
+
+        return new NativeHttpClient($defaultOptions, $maxHostConnections);
+    }
```

Comment:
What do you think of
```php
/**
      * @param array $defaultOptions     Default requests' options
      * @param int   $maxHostConnections The maximum number of connections to a single host
      */
    public static function createApi(array $defaultOptions = [], int $maxHostConnections = 6): ApiClientInterface
    {
        return new ApiClient(static::create($defaultOptions, $maxHostConnections));
    }
```
?

Reply:
Not sure it's useful. But can be discussed and done later in another PR!

Reply:
OK, added as `ApiClient::create()` :)